libs/application/template-api-modules/src/lib/modules/templates/secondary-school/secondary-school.service.ts

```diff
@@ -0,0 +1,224 @@
+
+    if (!canCreate) {
+      throw new TemplateApiError(
+        {
+          title: error.errorValidateCanCreateTitle,
+          summary: error.errorValidateCanCreateDescription,
+        },
+        400,
+      )
+    }
+  }
+
+  async deleteApplication({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<void> {
+    // If we are past the registration date for any of the selected programs, dont allow delete
+    if (
+      getEndOfDayUTCDate(getFirstRegistrationEndDate(application.answers)) <
+      new Date()
+    ) {
+      throw new TemplateApiError(
+        {
+          title: error.errorDeletePastRegistrationEndTitle,
+          summary: error.errorDeletePastRegistrationEndDescription,
+        },
+        500,
+      )
+    }
+
+    // Delete the application
+    await this.secondarySchoolClient.delete(auth, application.id)
+
+    // Send email to applicant and all contacts
+    await this.sendEmailAboutDeleteApplication(application)
+  }
+
+  private async sendEmailAboutDeleteApplication(
+    application: ApplicationWithAttachments,
+  ) {
+    const recipientList = getRecipients(application.answers)
+    for (let i = 0; i < recipientList.length; i++) {
+      if (recipientList[i].email) {
+        await this.sharedTemplateAPIService
+          .sendEmail(
+            (props) =>
+              generateApplicationRejectedEmail(props, recipientList[i]),
+            application,
+          )
+          .catch((e) => {
+            logger.error(
+              `Error sending email in deleteApplication for applicationID: ${application.id}`,
+              e,
+            )
+          })
+      }
+    }
+  }
+
+  async submitApplication({
+    application,
+    auth,
+  }: TemplateApiModuleActionProps): Promise<string> {
+    // Get values from answers
+    const applicant = getValueViaPath<SecondarySchoolAnswers['applicant']>(
+      application.answers,
+      'applicant',
+    )
+    const applicationType = getValueViaPath<SecondarySchoolApplicationType>(
+      application.answers,
+      'applicationType',
+    )
+    const contacts = getCleanContacts(application)
+    const schoolSelection = getCleanSchoolSelection(application)
+    const extraInformation = getValueViaPath<
+      SecondarySchoolAnswers['extraInformation']
+    >(application.answers, 'extraInformation')
+
+    // Submit the application
+    const applicationId = await this.secondarySchoolClient.create(auth, {
+      nationalId: auth.nationalId,
+      name: applicant?.name || '',
+      phone: applicant?.phoneNumber || '',
+      email: applicant?.email || '',
+      address: applicant?.address || '',
+      postalCode: applicant?.postalCode || '',
+      city: applicant?.city || '',
+      isFreshman: applicationType === SecondarySchoolApplicationType.FRESHMAN,
+      contacts: contacts,
+      schools: schoolSelection,
+      nativeLanguageCode: extraInformation?.nativeLanguageCode,
```

Comment:
Defaulta á 'is' (eða hvernig sem formattið er). Submitt failar ef notandi velur ekkert tungumál.